docs/source/basics/prompt_templates.rst

```diff
@@ -0,0 +1,249 @@
+.. _prompt_templates_usage_label:
+
+================
+Prompt Templates
+================
+
+Prompt templates are structured text templates which are used to format user prompts
+to optimize model performance on specific tasks. They can serve many purposes:
+
+1. Model-specific templates that are required whenever the model is prompted, such as the [INST]
+   tags in the instruct-tuned Llama2 and Mistral models. These models were pre-trained with these tags and using them
+   in inference can help ensure optimal performance.
+2. Task-specific templates to gear models for a particular task that it will expect after training.
+   Example include grammar correction (:class:`~torchtune.data.GrammarErrorCorrectionTemplate`),
+   summarization (:class:`~torchtune.data.SummarizeTemplate`), question answering (:class:`~torchtune.data.QuestionAnswerTemplate`),
+   and more.
+3. Community standardized templates, such as :class:`~torchtune.data.ChatMLTemplate`
```

Comment:
Sorry to sound like such a broken record, but just give examples for each of these. More showing, less talking about imo

Reply:
think examples for each of these is a bit too much for the intro section, esp when they can just click on the class and see the API ref